gateway/kickstart.sh

```diff
@@ -46,6 +46,8 @@ $SUDO docker run -d \
        -v /var/run/docker.sock:/var/run/docker.sock \
        -v /etc/passwd:/etc/passwd \
        -v /etc/group:/etc/group \
+       -v /var/run:/var/run \
```

Comment:
Older versions will also have to add this volume to the new container when doing the auto-update. See https://github.com/shellhub-io/shellhub/blob/master/agent/selfupdater/updater_docker.go#L83

E.g.
```
if v.LessThan(v0_4_2) {
  parent.info.HostConfig.Mounts = []mount.Mount{
    mount.Mount{
      Type:   mount.TypeBind,
      Source: "/var/run",
      Target: "/var/run",
    },
    mount.Mount{
      Type:   mount.TypeBind,
      Source: "/var/log",
      Target: "/var/log",
    },
  }
}
```

Or you can set `UtmpxFile  = "/host/var/run/utmp"` and `WtmpxFile  = "/host/var/log/wtmp"` when building using `docker` tag

Reply:
Good catch @gustavosbarreto; I completely forgot about this :-)

I have mixed feelings about the approach to take. We'd need to handle this on `kickstart.sh` on the gateway as well.

It seems adding the fine inside the `/host` volume is less hackish. We can later come up with a generic way to map to the host files to abstract this.

Reply:
I don't know how to set the file name constants differently when building using docker.  Guidance appreciated!

Reply:
@sixhills You can create a file for each platform type (docker and native) to set different values for `UtmpxFile` and `WtmpxFile`.

See:
https://github.com/shellhub-io/shellhub/blob/master/agent/init_docker.go
https://github.com/shellhub-io/shellhub/blob/master/agent/init_native.go

Reply:
`utmp_docker.go`:

```
// +build docker

const (
	UtmpxFile   = "/host/var/run/utmp"
	WtmpxFile   = "/host/var/log/wtmp"
)
```

`utmp_native.go`:

```
// +build !docker

const (
	UtmpxFile   = "/var/run/utmp"
	WtmpxFile   = "/var/log/wtmp"
)
```

Reply:
Thanks - I understand now,

Reply:
I don't think this will work - unless I'm misunderstanding, which is quite possible.

Despite using `-v /:/host` in the `docker run` command, the agent inside docker reports:

`level=warning msg="Open failed" err="open /host/var/run/utmp: no such file or directory" file=/host/var/run/utmp`

Does `docker  run -v ...` mount only the specified directory and not the tree beneath it?

Reply:
@sixhills what host distribution are you using?

Reply:
... and please post the output of:
```sh
mount
```

Reply:
Raspberry Pi OS (aka Raspbian).  I can try others if there may be differences.  Mount output follows:

dev/mmcblk0p2 on / type ext4 (rw,noatime)
devtmpfs on /dev type devtmpfs (rw,relatime,size=1827800k,nr_inodes=97773,mode=755)
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
proc on /proc type proc (rw,relatime)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,nodev,mode=755)
tmpfs on /run/lock type tmpfs (rw,nosuid,nodev,noexec,relatime,size=5120k)
tmpfs on /sys/fs/cgroup type tmpfs (ro,nosuid,nodev,noexec,mode=755)
cgroup2 on /sys/fs/cgroup/unified type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,name=systemd)
none on /sys/fs/bpf type bpf (rw,nosuid,nodev,noexec,relatime,mode=700)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_cls,net_prio)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=28,pgrp=1,timeout=0,minproto=5,maxproto=5,direct)
sunrpc on /run/rpc_pipefs type rpc_pipefs (rw,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,relatime)
mqueue on /dev/mqueue type mqueue (rw,relatime)
configfs on /sys/kernel/config type configfs (rw,relatime)
/dev/mmcblk0p1 on /boot type vfat (rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,errors=remount-ro)
tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,size=391976k,mode=700,uid=1000,gid=1000)
gvfsd-fuse on /run/user/1000/gvfs type fuse.gvfsd-fuse (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
fusectl on /sys/fs/fuse/connections type fusectl (rw,relatime)
nsfs on /run/docker/netns/default type nsfs (rw)

Reply:
@sixhills What is the output of `readlink -f /var/run/utmp`?

Reply:
That's a good question.  `readlink -f /var/run/utmp` reports: `/run/utmp`.

Reply:
Sorry this is proving rather more complicated than expected. I don't want to use too much of your time on this.

Reply:
Ok; so we won't be capable of handling it using the `/host` volume. You'll need to add `-v /var/run/utmp` on upgrade and installation script :-(

Reply:
OK, thanks.

Reply:
You need to handle the upgrade case as well. See https://github.com/shellhub-io/shellhub/pull/582/commits/505733e6843ed619f92f18cef2c0f177caf222c3#r551440473